sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/SqlOperators.java

```diff
@@ -72,6 +82,23 @@ public static SqlFunction createSimpleSqlFunction(String name, SqlTypeName retur
         SqlFunctionCategory.USER_DEFINED_FUNCTION);
   }
 
+  public static SqlUserDefinedAggFunction createUdafOperator(
+      String name,
+      SqlReturnTypeInference returnTypeInference,
+      AggregateFunction function) {
+    return new SqlUserDefinedAggFunction(
```

Comment:
Nice. By doing so it becomes a extensible way to define other AGG functions that Calcite does not have.

Glad you figured it out.